modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/TargetRequest.java

```diff
@@ -141,106 +140,87 @@ public void start(NHttpClientConnection conn) throws IOException, HttpException
+                            && Boolean.TRUE.toString().equals(requestMsgCtx.getProperty(NhttpConstants.POST_TO_URI))) {
+                        path = _url.toString();
+                    } else {
+                        path = _url.getPath()
+                                + ((_url.getQuery() != null && !_url.getQuery().isEmpty())
+                                        ? ("?" + _url.getQuery())
+                                        : "");
+                    }
+
+                }
+                headers.remove(HTTP.CONTENT_TYPE);
+            }
+        }
         
-		Object o = requestMsgCtx.getProperty(MessageContext.TRANSPORT_HEADERS);
-		if (o != null && o instanceof TreeMap) {
-			Map _headers = (Map) o;
-			String trpContentType = (String) _headers.get(HTTP.CONTENT_TYPE);
-			if (trpContentType != null && !trpContentType.equals("")) {
-				if (!TargetRequestFactory.isMultipartContent(trpContentType)) {
-					addHeader(HTTP.CONTENT_TYPE, trpContentType);
-				}
-			}
-
-		}
+        Object o = requestMsgCtx.getProperty(MessageContext.TRANSPORT_HEADERS);
+        if (o != null && o instanceof TreeMap) {
+            Map _headers = (Map) o;
+            String trpContentType = (String) _headers.get(HTTP.CONTENT_TYPE);
+            if (trpContentType != null && !trpContentType.equals("")) {
+                if (!TargetRequestFactory.isMultipartContent(trpContentType)) {
+                    addHeader(HTTP.CONTENT_TYPE, trpContentType);
+                }
+            }
+
+        }
+                                                            
+        request = new BasicHttpEntityEnclosingRequest(method, path,
+                version != null ? version : HttpVersion.HTTP_1_1);
         
+        BasicHttpEntity entity = new BasicHttpEntity();
         
-  
-                                                            
-        if (hasEntityBody) {
-            request = new BasicHttpEntityEnclosingRequest(method, path,
-                    version != null ? version : HttpVersion.HTTP_1_1);
-
-            BasicHttpEntity entity = new BasicHttpEntity();
-
-
-            boolean forceContentLength = requestMsgCtx.isPropertyTrue(
-                                                                   NhttpConstants.FORCE_HTTP_CONTENT_LENGTH);
-            boolean forceContentLengthCopy = requestMsgCtx.isPropertyTrue(
-                                                                   PassThroughConstants.COPY_CONTENT_LENGTH_FROM_INCOMING);
-                                
-            if (forceContentLength) {
-                entity.setChunked(false);
-                if (forceContentLengthCopy && contentLength > 0) {
-                    entity.setContentLength(contentLength);
-                } 
-            }else{
-             if (contentLength != -1) {
+        boolean forceContentLength = requestMsgCtx.isPropertyTrue(NhttpConstants.FORCE_HTTP_CONTENT_LENGTH);
```

Comment:
seems like you forcefully set the FORCE_HTTP_CONTENT_LENGTH and COPY_CONTENT_LENGTH_FROM_INCOMING in here if it is set in here, It will not affect the scenarios like. If the content length is 0, then It will set the chunking header.
 if (contentLength != -1) {
                 --------
             } else {
                 entity.setChunked(chunk);
             }

So we can set the properties in main thread, as in bellow

private void **sendRequestContent**(final MessageContext msgContext) throws AxisFault {
	boolean forceContentLength = msgContext.isPropertyTrue(NhttpConstants.FORCE_HTTP_CONTENT_LENGTH);
	    boolean forceContentLengthCopy = msgContext.isPropertyTrue(PassThroughConstants.COPY_CONTENT_LENGTH_FROM_INCOMING);

		if (forceContentLength && forceContentLengthCopy && msgContext.getProperty(PassThroughConstants.ORGINAL_CONTEN_LENGTH) != null) {
			 msgContext.setProperty(PassThroughConstants.PASSTROUGH_MESSAGE_LENGTH, Long.parseLong((String)msgContext.getProperty(PassThroughConstants.ORGINAL_CONTEN_LENGTH) ));
		}